File: core/src/main/java/info/nightscout/androidaps/plugins/iob/iobCobCalculator/AutosensResult.java
package info.nightscout.androidaps.plugins.iob.iobCobCalculator;

import org.json.JSONException;
import org.json.JSONObject;
import org.slf4j.LoggerFactory;

import info.nightscout.androidaps.logging.LTag;

/**
 * Created by mike on 06.01.2017.
 */
public class AutosensResult {

    //default values to show when autosens algorithm is not called
    public double ratio = 1d;
    public double carbsAbsorbed = 0d;
    public String sensResult = "autosens not available";
    public String pastSensitivity = "";
    public String ratioLimit = "";

    public JSONObject json() {
        JSONObject ret = new JSONObject();
        try {
            ret.put("ratio", ratio);
            ret.put("ratioLimit", ratioLimit);
            ret.put("pastSensitivity", pastSensitivity);
            ret.put("sensResult", sensResult);
            ret.put("ratio", ratio);
        } catch (JSONException e) {
            LoggerFactory.getLogger(LTag.CORE.getTag()).error("Unhandled exception", e);
        }
        return ret;
    }

    @Override
    public String toString() {
        return json().toString();
    }
}

File: core/interfaces/src/main/kotlin/app/aaps/core/interfaces/aps/AutosensResult.kt
package app.aaps.core.interfaces.aps

import kotlinx.serialization.Serializable

@Serializable
data class AutosensResult(

    //default values to show when autosens algorithm is not called
    var ratio: Double = 1.0,
    var carbsAbsorbed: Double = 0.0,
    var sensResult: String = "autosens not available",
    var pastSensitivity: String = "",
    var ratioLimit: String = "",
    // Dynamic sensitivity
    // insulin within last24h/average7days
    var ratioFromTdd: Double = 1.0,
    // carbs within last24h/average7days
    var ratioFromCarbs: Double = 1.0
)